Mathlib/Geometry/Manifold/VectorBundle/LocalFrame.lean

```diff
@@ -283,3 +301,334 @@ end
+lemma contMDiffOn_localFrame_baseSet
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e] (b : Basis ι 𝕜 F) (i : ι) :
+    CMDiff[e.baseSet] n (T% (b.localFrame e i)) := by
+  rw [e.contMDiffOn_section_baseSet_iff]
+  apply (contMDiffOn_const (c := b i)).congr
+  intro y hy
+  simp [localFrame, hy, localFrame_toBasis_at]
+
+omit [IsManifold I 0 M] in
+variable (I) in
+/-- `b.localFrame e i` is indeed a local frame on `e.baseSet` -/
+lemma localFrame_isLocalFrameOn_baseSet
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e] (b : Basis ι 𝕜 F) : IsLocalFrameOn I F n (b.localFrame e) e.baseSet
+    where
+  contMDiffOn i := b.contMDiffOn_localFrame_baseSet _ e i
+  linearIndependent := by
+    intro x hx
+    convert (b.localFrame_toBasis_at e hx).linearIndependent
+    simp [localFrame, hx, localFrame_toBasis_at]
+  generating := by
+    intro x hx
+    convert (b.localFrame_toBasis_at e hx).span_eq.ge
+    simp [localFrame, hx, localFrame_toBasis_at]
+
+omit [IsManifold I 0 M] in
+lemma _root_.contMDiffAt_localFrame_of_mem
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e] (b : Basis ι 𝕜 F) (i : ι) (hx : x ∈ e.baseSet) :
+    CMDiffAt n (T% (b.localFrame e i)) x :=
+  (b.localFrame_isLocalFrameOn_baseSet I n e).contMDiffAt e.open_baseSet hx _
+
+@[simp]
+lemma localFrame_apply_of_mem_baseSet
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e] (b : Basis ι 𝕜 F) {i : ι} (hx : x ∈ e.baseSet) :
+    b.localFrame e i x = b.localFrame_toBasis_at e hx i := by
+  simp [localFrame, hx]
+
+@[simp]
+lemma localFrame_apply_of_notMem
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e] (b : Basis ι 𝕜 F) {i : ι} (hx : x ∉ e.baseSet) :
+    b.localFrame e i x = 0 := by
+  simp [localFrame, hx]
+
+lemma localFrame_toBasis_at_coe
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e] (b : Basis ι 𝕜 F) (i : ι) (hx : x ∈ e.baseSet) :
+    b.localFrame_toBasis_at e hx i = b.localFrame e i x := by simp [hx]
+
+variable [ContMDiffVectorBundle 1 F V I]
+
+open scoped Classical in
+variable (I) in
+/-- Coefficients of a section `s` of `V` w.r.t. the local frame `b.localFrame e i`.
+
+If x is outside of `e.baseSet`, this returns the junk value 0. -/
+noncomputable def localFrame_coeff
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e]
+    (b : Basis ι 𝕜 F) (i : ι) : (Π x : M, V x) →ₗ[𝕜] M → 𝕜 :=
+  (b.localFrame_isLocalFrameOn_baseSet I 1 e).coeff i
+
+variable {e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M)}
+  [MemTrivializationAtlas e] {b : Basis ι 𝕜 F} {x : M}
+
+omit [IsManifold I 0 M] in
+variable (e b) in
+@[simp]
```

Comment:
Again, I'm dubious about these conditional `simp` lemmas. (I'll stop raising this point; perhaps they're a good idea.)